GitUI/CommandsDialogs/FormFileHistory.cs

```diff
@@ -468,6 +469,31 @@ private void FileHistoryContextMenuOpening(object sender, CancelEventArgs e)
             manipulateCommitToolStripMenuItem.Enabled =
                 selectedRevisions.Count == 1 && !selectedRevisions[0].IsArtificial;
             saveAsToolStripMenuItem.Enabled = selectedRevisions.Count == 1;
+
+            // clipboard branch and tag menu handling
+            {
+                var revision = selectedRevisions[0];
+                copyToClipboardToolStripMenuItem.Enabled = !revision.IsArtificial;
+
+                if (copyToClipboardToolStripMenuItem.Enabled)
+                {
+                    var gitRefListsForRevision = new UserControls.RevisionGridClasses.GitRefListsForRevision(revision);
```

Comment:
This introduces a coupling to `RevisionGrid` specific implementations, which is unacceptable. 
Neither implementation, which this dependency is referenced for, is complex to justify the coupling. Please inline "get branches" and "get tags"